QC2 Tone Paging

DeathSoulYTDev opened the pull request:
- Fixed QC2 Tone Paging
- Added Support For Stacked QC2 Tones Paging

DeathSoulYTDev committed:
Stacked QC2 Tone Paging

- Added Support for Stacked QC2 Tone Paging and Fixed the Pager Not Alerting to Tones

DeathSoulYTDev committed:
Update client.js

Firav reviewed (changes requested):
Looks like you've been working on an outdated fork of this project, as this PR removes many recent changes that have been made in recent updates. Please merge your fork with the most recent commits of this main, verify that your changes are still valid, and make those changes to this PR.

DeathSoulYTDev committed:
Updated Code

- Updated code to Include the New Updates Made to WH5M in since QC2 Paging was updated

DeathSoulYTDev commented on client/ui/js/script.js:
```diff
@@ -1520,63 +1531,123 @@ function processTone(frequency) {
     }
 }
 
+// function detectQC2Pair() {
```
forgot to remove them my bad will remove and reupload

DeathSoulYTDev committed:
Comments Removal

- Removed Old Commented out Code that was not being used

Firav reviewed (commented):
This looks good now on face value. Would still need to test these changes within a fivem env to ensure feature validation.

firealarmss commented:
Does it work? @Firav

Firav commented:
> Does it work? @Firav

It works-ish. Works fine with tones higher than 320Hz. Anything 320 and below it does not register. I had the qc set to 123x321 in the codeplug and this is what spat back in console. (Which looking at the code, looks like the valid range is 300 to 3000. If we keep it like that, we'll just have to add some documentation for it.) However for the tones in-range, 9/10 of the tones sent from console registered as a page. The one that didnt register is when a starttime was adjusted real wacky by the FiveM script.

```
connected as conv peer, aff restrictions will be ignored (@whackerlinkfivem/client/ui/js/script.js:1052)
WebSocket connection established (@whackerlinkfivem/client/ui/js/script.js:1057)
Adjusting startTime: 7 -> 23.490666666666666 (@whackerlinkfivem/client/ui/js/PcmPlayer.js:152)
checking pair: A=320 Hz (0 ms), B=320 (2641 ms) (@whackerlinkfivem/client/ui/js/script.js:1555)
checking pair: A=320 Hz (2641 ms), B=320 (1 ms) (@whackerlinkfivem/client/ui/js/script.js:1555)
checking pair: A=320 Hz (1 ms), B=320 (3 ms) (@whackerlinkfivem/client/ui/js/script.js:1555)
Adjusting startTime: 43.141333333333336 -> 51.49066666666667 (@whackerlinkfivem/client/ui/js/PcmPlayer.js:152)
checking pair: A=320 Hz (3 ms), B=320 (2592 ms) (@whackerlinkfivem/client/ui/js/script.js:1555)
Adjusting startTime: 54.101333333333336 -> 62.989333333333335 (@whackerlinkfivem/client/ui/js/PcmPlayer.js:152)
checking pair: A=320 Hz (2592 ms), B=320 (2671 ms) (@whackerlinkfivem/client/ui/js/script.js:1555)
Adjusting startTime: 65.66933333333333 -> 98.50133333333333 (@whackerlinkfivem/client/ui/js/PcmPlayer.js:152)
checking pair: A=320 Hz (2671 ms), B=320 (2698 ms) (@whackerlinkfivem/client/ui/js/script.js:1555)
```

DeathSoulYTDev commented:
> > Does it work? @Firav
> 
> 
> 
> It works-ish. Works fine with tones higher than 320Hz. Anything 320 and below it does not register. I had the qc set to 123x321 in the codeplug and this is what spat back in console. (Which looking at the code, looks like the valid range is 300 to 3000. If we keep it like that, we'll just have to add some documentation for it.) However for the tones in-range, 9/10 of the tones sent from console registered as a page. The one that didnt register is when a starttime was adjusted real wacky by the FiveM script.
> 
> 
> 
> ```
> 
> connected as conv peer, aff restrictions will be ignored (@whackerlinkfivem/client/ui/js/script.js:1052)
> 
> WebSocket connection established (@whackerlinkfivem/client/ui/js/script.js:1057)
> 
> Adjusting startTime: 7 -> 23.490666666666666 (@whackerlinkfivem/client/ui/js/PcmPlayer.js:152)
> 
> checking pair: A=320 Hz (0 ms), B=320 (2641 ms) (@whackerlinkfivem/client/ui/js/script.js:1555)
> 
> checking pair: A=320 Hz (2641 ms), B=320 (1 ms) (@whackerlinkfivem/client/ui/js/script.js:1555)
> 
> checking pair: A=320 Hz (1 ms), B=320 (3 ms) (@whackerlinkfivem/client/ui/js/script.js:1555)
> 
> Adjusting startTime: 43.141333333333336 -> 51.49066666666667 (@whackerlinkfivem/client/ui/js/PcmPlayer.js:152)
> 
> checking pair: A=320 Hz (3 ms), B=320 (2592 ms) (@whackerlinkfivem/client/ui/js/script.js:1555)
> 
> Adjusting startTime: 54.101333333333336 -> 62.989333333333335 (@whackerlinkfivem/client/ui/js/PcmPlayer.js:152)
> 
> checking pair: A=320 Hz (2592 ms), B=320 (2671 ms) (@whackerlinkfivem/client/ui/js/script.js:1555)
> 
> Adjusting startTime: 65.66933333333333 -> 98.50133333333333 (@whackerlinkfivem/client/ui/js/PcmPlayer.js:152)
> 
> checking pair: A=320 Hz (2671 ms), B=320 (2698 ms) (@whackerlinkfivem/client/ui/js/script.js:1555)
> 
> ```

I will look into seeing if I can get the lower tone sets to also work as I had a few set in the 100-200 range and it works at a 50/50 rate

firealarmss commented:
If I recall, the lowest valid QC2 frequency is like 309

Firav commented:
Then those limits set in the code should be fine then

firealarmss commented on client/ui/js/script.js:
```diff
@@ -1520,60 +1531,44 @@ function processTone(frequency) {
-                    console.log(`QC2 ALERT: A=${pair.a} B=${pair.b}`);
-                    minitorStandard();
+                    console.log(`QC2 Match: A=${pair.a}, B=${pair.b}`);
+                    minitorStandard();  // ✅ Start paging workflow
```
Why is there an emoji?

DeathSoulYTDev replied:
because i felt like being dumb and added it as i like to do stupid things at times and that was one of those times LOL

firealarmss commented on client/ui/js/script.js:
```diff
@@ -1753,12 +1748,53 @@ function buttonBeep() {
+
+    const nextTone = tonesQueue.shift();
+    if (!nextTone) {
+        isPaging = false; // ✅ Done with queue
```
Emoji

Firav commented:
HAHAHA I had'nt noticed those. Gotta love AI comments. It loves its emojis

DeathSoulYTDev commented:
> HAHAHA I had'nt noticed those. Gotta love AI comments. It loves its emojis

Actually that was me just trying to be funny LMFAO

Firav commented:
> > HAHAHA I had'nt noticed those. Gotta love AI comments. It loves its emojis
> 
> Actually that was me just trying to be funny LMFAO

Can you please remove those emoji's from the source code so that its more compliant?

DeathSoulYTDev commented:
> > > HAHAHA I had'nt noticed those. Gotta love AI comments. It loves its emojis
> 
> > 
> 
> > Actually that was me just trying to be funny LMFAO
> 
> 
> 
> Can you please remove those emoji's from the source code so that its more compliant?

Yes give me a little been pretty busy with work

DeathSoulYTDev committed:
Update script.js

DeathSoulYTDev commented:
Emojis have been removed

Firav commented on client/ui/js/script.js:
```diff
@@ -1520,60 +1531,44 @@ function processTone(frequency) {
-                    console.log(`QC2 ALERT: A=${pair.a} B=${pair.b}`);
-                    minitorStandard();
+                    console.log(`QC2 Match: A=${pair.a}, B=${pair.b}`);
+                    minitorStandard();  // ✅ Start paging workflow
```
You forgot to remove this emoji.

DeathSoulYTDev replied:
That emoji has been removed

Firav commented on client/ui/js/script.js:
```diff
@@ -151,7 +156,7 @@ function isMobile() {
 
 function isScannerModel() {
-    return radioModel === "UNIG5";
+    return radioModel === "UNIG5" || radioModel === "MIN6";
```
@DeathSoulYTDev Do you plan on implementing a MIN6 model at some point?

DeathSoulYTDev replied:
yes i do just got to find the time to sit and do it

DeathSoulYTDev committed:
Merge branch 'WhackerLink:main' into main

Firav reviewed (changes requested):
Not seeing any new changes

DeathSoulYTDev committed:
Remove checkmark emoji from comment

Firav reviewed (approved):
Looks good for merge @firealarmss

firealarmss commented:
Confirming again @Firav that this is good for merge?

Firav commented:
> Confirming again @Firav that this is good for merge?

Yeah should be good
